lms/djangoapps/shoppingcart/models.py

```diff
@@ -836,6 +903,7 @@ class CourseRegistrationCode(models.Model):
     created_at = models.DateTimeField(default=datetime.now(pytz.utc))
     order = models.ForeignKey(Order, db_index=True, null=True, related_name="purchase_order")
     invoice = models.ForeignKey(Invoice, null=True)
+    invoice_item = models.ForeignKey(CourseRegistrationCodeInvoiceItem, null=True)
```

Comment:
hrm. Not sure if this is going to bite us having both the options to have a FK to both invoice and invoice_item. Will invoice and invoice_item FK's always be populated? If so, it's a bit redundant, but I guess we need it to maintain backwards compatibility with existing data. Also, we'd always need to assert that the InvoiceItem also points to the same FK as invoice here.

That said, do we have good test coverage for situations where